src/tests/ftest/datamover/dm_dst_create.py

```diff
@@ -0,0 +1,272 @@
+                cont.pool, cont, flags=self.ior_flags[1])
+        else:
+            self.dataset_verify(cont, 1, 1, 1, 0, [1024], [])
+
+        if check_attr_prop:
+            cont.open()
+            self.verify_cont_prop(cont, prop_list)
+            self.verify_usr_attr(cont)
+            cont.close()
+
+    def get_cont_prop(self, cont):
+        """Get all container properties with daos command.
+
+        Args:
+            cont (TestContainer): the container to get props of.
+
+        Returns:
+            list: string list of properties from daos command.
+
+        """
+        prop_result = self.daos_cmd.container_get_prop(
+            cont.pool.uuid, cont.uuid)
+        prop_text = prop_result.stdout_text
+        prop_list = prop_text.split('\n')[1:]
+        return prop_list
+
+    def verify_cont_prop(self, cont, prop_list):
+        """Verify container properties against an input list.
+        Expects the container to be open.
+
+        Args:
+            cont (TestContainer): the container to verify.
+            prop_list (list): list of properties from get_cont_prop.
+
+        """
+        actual_list = self.get_cont_prop(cont)
+
+        # Make sure sizes match
+        if len(prop_list) != len(actual_list):
+            self.log.info("Expected\n%s\nbut got\n%s\n",
+                          prop_list, actual_list)
+            self.fail("Container property verification failed.")
+
+        # Make sure each property matches
+        for prop_idx, prop in enumerate(prop_list):
+            # This one is not set
+            if cont.type.value == "POSIX" and "OID" in prop_list[prop_idx]:
+                continue
+            if prop != actual_list[prop_idx]:
+                self.log.info("Expected\n%s\nbut got\n%s\n",
+                              prop_list, actual_list)
+                self.fail("Container property verification failed.")
+
+        self.log.info("Verified %d container properties:\n%s",
+                      len(actual_list), actual_list)
+
+    @staticmethod
+    def get_cont_usr_attr():
+        """Generate some user attributes"""
+        attrs = {}
+        attrs["attr1".encode("utf-8")] = "value 1".encode("utf-8")
+        attrs["attr2".encode("utf-8")] = "value 2".encode("utf-8")
+        return attrs
+
+    def verify_usr_attr(self, cont):
+        """Verify user attributes. Expects the container to be open.
+
+        Args:
+            cont (TestContainer): the container to verify.
+
+        """
+        attrs = self.get_cont_usr_attr()
+        actual_attrs = cont.container.get_attr(list(attrs.keys()))
+
+        # Make sure the sizes match
+        if len(attrs.keys()) != len(actual_attrs.keys()):
+            self.log.info("Expected\n%s\nbut got\n%s\n",
+                          attrs.items(), actual_attrs.items())
+            self.fail("Container user attributes verification failed.")
+
+        # Make sure each attr matches
+        for attr in attrs.keys():
```

Comment:
(pylint-consider-iterating-dictionary) Consider iterating the dictionary directly instead of calling .keys()